### PR TITLE
docs: Improve Dialog navigation documentation

### LINKED ISSUE #1672: [Navigation] Modal navigation is being dismissed on windows when alt+tab
## Current behavior
If you navigate to a page using the modal navigation (`!`), when you alt+tab the page is being dismissed automatically.

## Expected behavior
For me a "modal" nav should even prevent user to dismiss the page, the page should not close itself.

> "Users must interact with the modal window before they can return to the parent window."
> [Modal window (wikipedia)](https://en.wikipedia.org/wiki/Modal_window)

## How to reproduce it (as minimally and precisely as possible)
Navigate to a page using the `!` like in [this repro](https://github.com/unoplatform/uno.extensions/files/12018320/unoExtensionBasicApp.zip) (from https://github.com/unoplatform/uno/issues/12871)

## Environment
Nuget Package:

Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renders for Xamarin.Forms
- [x] Windows
- [ ] Build tasks

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

## Anything else we need to know?
The fact it works on other platform that are using uno is a bug https://github.com/unoplatform/uno/issues/12928

### PR DESCRIPTION
**Pull request by eriklimakc**
GitHub Issue (If applicable): closes #1672

## PR Type

What kind of change does this PR introduce?
- Documentation content changes

## What is the current behavior?
Current docs can lead people to believe that the "!" qualifier means **Modal**, where it actually means **Dialog**, and the behavior (modal X flyout) is set by the view type (`ContentDialog` X `Page`).

## What is the new behavior?

Clarification among Dialog X Modal X Flyout.

### CODE REVIEW

File: doc/Learn/Navigation/HowTo-ShowDialog.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+uid: Uno.Extensions.Navigation.HowToShowDialog
+---
+# How-To: Display a Dialog (modal or flyout)
+
+This topic provides a step-by-step guide on using Navigation to display a Dialog.
+
+## Step-by-steps
+
+> [!IMPORTANT]
+> This guide assumes you used the template wizard or `dotnet new unoapp` to create your solution. If not, it is recommended that you follow the [Creating an application with Uno.Extensions article](xref:Uno.Extensions.HowToGettingStarted) for creating an application from the template.
```
Comment (agneszitte) [inline doc/Learn/Navigation/HowTo-ShowDialog.md:11]:
```suggestion
> This guide assumes you used the template wizard or `dotnet new unoapp` to create your solution. If not, it is recommended that you follow the [Creating an application with Uno.Extensions article](xref:Uno.Extensions.HowToGettingStarted) to create an application from the template.
```

File: doc/Learn/Navigation/HowTo-ShowDialog.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        <Grid>
+            <TextBlock Text="Dialog Navigation"
+                       FontSize="32"
+                       HorizontalAlignment="Center"
+                       VerticalAlignment="Center" />
+        </Grid>
+    </Page>
+    ```
+
+- Update the `Button` in `MainPage.xaml` as follows, which wires up the `Click` event to the `ShowDialogClick` method
```
Comment (morning4coffe-dev) [inline doc/Learn/Navigation/HowTo-ShowDialog.md:36]:
```suggestion
- Update the `Button` in `MainPage.xaml` as follows, which wires up the `Click` event to the `ShowDialogClick` method:
```

File: doc/Learn/Navigation/HowTo-ShowDialog.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    ```xml
+    <Button Content="Show flyout"
+            Click="ShowDialogClick"
+            Grid.Row="1"
+            HorizontalAlignment="Center"
+            VerticalAlignment="Center"/>
+    ```
+
+- Add the `ShowDialogClick` method to the `MainPage.xaml.cs` file
```
Comment (morning4coffe-dev) [inline doc/Learn/Navigation/HowTo-ShowDialog.md:46]:
```suggestion
- Add the `ShowDialogClick` method to the `MainPage.xaml.cs` file:
```

File: doc/Learn/Navigation/HowTo-ShowDialog.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                HorizontalAlignment="Center"
+                VerticalAlignment="Center">
+        <Button Content="Show flyout"
+                Click="ShowDialogClick" />
+        <Button Content="Show flyout from XAML"
+                uen:Navigation.Request="!Sample" />
+    </StackPanel>
+    ```
+
+## Flyout X Modal
```
Comment (morning4coffe-dev) [inline doc/Learn/Navigation/HowTo-ShowDialog.md:70]:
```suggestion
## Flyout vs. Modal
```

**Review by morning4coffe-dev — APPROVED** [review]

**Comment by eriklimakc** [comment]
Rebased to include https://github.com/unoplatform/uno.extensions/pull/2343

File: doc/Learn/Navigation/HowTo-ShowDialog.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+This topic provides a step-by-step guide on using Navigation to display a Dialog.
+
+## Step-by-steps
+
+> [!IMPORTANT]
+> This guide assumes you used the template wizard or `dotnet new unoapp` to create your solution. If not, it is recommended that you follow the [**Creating an application with Uno.Extensions** documentation](xref:Uno.Extensions.HowToGettingStarted) to create an application from the template.
+
+### 1. Displaying a dialog from code
+
+- Add new `Page`, `SamplePage.xaml`, which will be used to display content inside the flyout.
```
Comment (kazo0) [inline doc/Learn/Navigation/HowTo-ShowDialog.md:15]:
```suggestion
- Add a new `Page`, `SamplePage.xaml`, which will be used to display content inside the flyout.
```

**Review by kazo0 — APPROVED** [review]

**Review by nickrandolph — APPROVED** [review]

**Comment by agneszitte** [comment]
@Mergifyio backport release/stable/4.1

**Pull request merged**